hyko_toolkit/models/similarity_search/Dockerfile

```diff
@@ -5,7 +5,8 @@ WORKDIR /app
 RUN --mount=type=cache,target=/root/.cache/pip pip install langchain 
 RUN --mount=type=cache,target=/root/.cache/pip pip install sentence_transformers
 RUN --mount=type=cache,target=/root/.cache/pip pip install faiss-cpu
-
+RUN --mount=type=cache,target=/root/.cache/pip pip install rank_bm25
```

Comment:
move these extra deps to only the function that require it

Reply:
make sure to have minimal dependencies in each docker file, only what is needed to run that model